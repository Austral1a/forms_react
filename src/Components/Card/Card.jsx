import React from 'react'
import classNames from 'classnames/dedupe'
import './Card.scss'

export function Card({
    style,
    className,
    children
}) {
    const customClasses = classNames('card', className)
    return (
        <div
            style={style}
            className={customClasses}
        >
            {children}
        </div>
    )
}
