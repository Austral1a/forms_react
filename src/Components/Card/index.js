export * from './Card.scss'
export * from './Card'
