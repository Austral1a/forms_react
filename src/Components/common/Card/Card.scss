@import '../../../utils/variables';

.card {
  display: flex;
  border-radius: 4px;
  background-color: $card-bg-color;
  padding: 8px 12px;
  width: 135px;
  height: 35px;
  text-align: center;
  align-items: center;
  justify-content: center;
}
