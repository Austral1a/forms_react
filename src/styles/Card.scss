.card {
  border-radius: 4px;
  background-color: darkkhaki;
}
