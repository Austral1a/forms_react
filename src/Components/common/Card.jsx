import React from 'react'
import '../../styles/Card.scss'

function Card({
    style,
    className,
    children
}) {
    return (
        <div
            style={style}
            className={`card ${className}`}
        >
            {children}
        </div>
    )
}

export default Card
