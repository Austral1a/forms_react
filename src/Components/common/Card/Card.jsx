import React from 'react'
import classNames from 'classnames/dedupe'
import './Card.scss'


function Card({
    style,
    className,
    children
}) {
    const customClasses = classNames('card', className)
    return (
        <div
            style={style}
            className={customClasses}
        >
            {children}
        </div>
    )
}

export default Card
