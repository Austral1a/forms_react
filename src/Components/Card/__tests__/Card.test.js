import {Card} from '../Card'
import {shallow} from 'enzyme'
import toJson from 'enzyme-to-json'
describe('Card Component' ,() => {
    it('Card must match snapshot', () => {
        const wrapper = shallow(<Card />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    it("must set style", () => {
        const bgColor = '#fff'
        const wrapper = shallow(<Card style={{backgroundColor: bgColor}} />)

        expect(wrapper.find('.card').prop('style').backgroundColor).toBe(bgColor)
    })

    it("if className prop exists must set additional className", () => {
        const className = 'card-first'
        const wrapper = shallow(<Card className={className} />)

        expect(wrapper.find('.card').prop('className')).toBe(`card ${className}`)
    })

    it('if className prop is not exists must not set additional className', () => {
        const wrapper = shallow(<Card />)

        expect(wrapper.find('.card').prop('className')).toBe('card')
        expect(wrapper.find('.card').prop('className')).not.toBe('card ')
    })

    it("must render children", () => {
        const text = 'child'
        const wrapper = shallow(<Card><p>{text}</p></Card>)

        expect(wrapper.find('p')).toHaveLength(1)
        expect(wrapper.find('p').text()).toBe(text)
        expect(wrapper.find('p').text()).not.toBe(text + ' ')
    })
})