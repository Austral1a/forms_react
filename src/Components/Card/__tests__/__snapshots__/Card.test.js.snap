// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component Card must match snapshot 1`] = `
<div
  className="card"
/>
`;
